pdns/dnsparser.cc

```diff
@@ -685,6 +686,23 @@ void PacketReader::xfrSvcParamKeyVals(set<SvcParam> &kvs) {
       kvs.insert(SvcParam(key, blob));
       break;
     }
+    case SvcParam::tls_supported_groups: {
+      if (len % 2 != 0) {
+        throw std::out_of_range("invalid length for " + SvcParam::keyToString(key));
+      }
+      vector<uint16_t> groups;
+      auto stop = d_pos + len;
+      while (d_pos < stop)
```

Comment:
How about not making this an infinite loop...
```suggestion
      for (auto stop = d_pos + len; d_pos < stop; d_pos += 2)
```

Reply:
Doh! Missed that, sorry.